### PR TITLE
Fixed #1105, case insensitive comparison of username and password

### LINKED ISSUE #1105: Usernames should be case insensitive when checked in password
1. Register
2. Create username in this example: bobby
3. Add password "bobby1234!" cannot save user because username is in password
4. Try "Bobby1234!" user is saved

We should be case insensitive when checking for usernames in passwords.

### PR DESCRIPTION
**Pull request by valaparthvi**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]
#1105 
-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Risks

[List any risks that could arise from merging your pull request.]

-

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
- [ ] **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
- [ ] **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 
- [ ] **If this is a bug fix, are tests for the issue in place**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: cadasta/accounts/forms.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def clean_password1(self):
         if email[0] in password:
             errors.append(_("Passwords cannot contain your email."))
 
-        if self.data.get('username') in password:
+        if self.data.get('username').casefold() in password.casefold():
```
Comment (oliverroick) [inline cadasta/accounts/forms.py]: Is there a particular reason why you used `casefold` instead of `lower`?

Comment (seav) [inline cadasta/accounts/forms.py]: Case folding is the recommended way to compare strings case-insensitively in the age of Unicode: https://www.w3.org/International/wiki/Case_folding

Comment (oliverroick) [inline cadasta/accounts/forms.py]: Ah ok fair enough. The Python docs did not explain it as good as the page you linked. We should stick to `casefold` then

**Comment by valaparthvi** [comment]
@oliverroick  I'll add necessary tests and submit a new PR in a new branch as you suggested. I might need help with the Javascript code, I'll let you know if I need help.

**Comment by oliverroick** [comment]
Thanks, can we close this PR then?

**Comment by valaparthvi** [comment]
@oliverroick Okay.

**Pull request closed without merging**